File: src/Systems/Fortran/patterns/broker.f90
module broker_example
  implicit none
contains
  logical function run()
    run = trim(route('inventory','sku-1')) == 'inventory:sku-1=7' .and. &
          trim(route('customer','17')) == 'customer:17=active'
  contains
    function route(service, key) result(text)
      character(len=*), intent(in) :: service, key
      character(len=96) :: text
      if (service == 'inventory') then
        text = 'inventory:'//trim(key)//'=7'
      else
        text = 'customer:'//trim(key)//'=active'
      end if
    end function
  end function
end module

File: src/Systems/Swift/patterns/Broker.swift
enum BrokerExample {
    static func run() -> Bool {
        let services: [String: (String) -> String] = [
            "inventory": { "inventory:\($0)=7" },
            "customer": { "customer:\($0)=active" }
        ]
        return services["inventory"]!("sku-1") == "inventory:sku-1=7"
            && services["customer"]!("17") == "customer:17=active"
    }
}
